### PR TITLE
Connection reset blog post

### PR DESCRIPTION
**Pull request by anfernee**
Blog: kube-proxy-subtleties-debugging-an-intermittent-connection-resets

### CODE REVIEW

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+topic.
+
+## The symptom
+
+We saw a user report seeing connection resets using a ClusterIP service to serve
+large files to pods running in their cluster. Initial debugging of the cluster
+did not yield anything interesting: network connectivity was fine and
+downloading the files did not hit any issues. However, when we ran the workload
+in parallel across many clients, we were able to reproduce the problem. Adding
+to the mystery was the fact that when the problem could not be reproduced when
```
Comment (DanyC97) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: i guess you want to remove `when` from `was the fact that when the problem `

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+I recently came across a bug that causes intermittent connection resets.  After
+some digging, I found it was caused by a subtle combination of several different
+network subsystems. It helps me understand Kubernetes networking better, and I
+think it’s worthwhile sharing to a wider audience who are interested in the same
+topic.
+
+## The symptom
+
+We saw a user report seeing connection resets using a ClusterIP service to serve
```
Comment (DanyC97) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: maybe we can reformat to say 

> We received a user report claiming connections getting resets while using a Kubernetes Service of type `ClusterIP` serving large files to pods ...

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+large files to pods running in their cluster. Initial debugging of the cluster
+did not yield anything interesting: network connectivity was fine and
+downloading the files did not hit any issues. However, when we ran the workload
+in parallel across many clients, we were able to reproduce the problem. Adding
+to the mystery was the fact that when the problem could not be reproduced when
+the workload was run using VMs without Kubernetes. There is a simple app to
+reproduce the issue [here](https://github.com/tcarmet/k8s-connection-reset). The
+problem clearly has something to do with Kubernetes networking, but what?
```
Comment (DanyC97) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: i would personally reformat it as

> The problem (which could be easily reproduced using [a simple app](https://github.com/tcarmet/k8s-connection-reset) clearly has something to do with Kubernetes networking, but what?

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+## Kubernetes networking basics
+
+Before digging into this problem, let’s talk a little bit about some basics of
+Kubernetes networking first. When a pod initializes a connection to either
+another pod, a in-cluster service or an external service, Kubernetes handles it
+differently.
```
Comment (DanyC97) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: i would personally reformat it as 

> Before digging into the problem, let’s talk a bit about basics of Kubernetes networking as the connection to either another pod, an in-cluster service or an external service is being handled differently by Kubernetes while the pod is being initialized.

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+the following rule in your cluster.
+
+```yaml apiVersion: extensions/v1beta1 kind: DaemonSet metadata: name:
+startup-script labels: app: startup-script spec: template: metadata: labels:
+app: startup-script spec: hostPID: true containers:
+      - name: startup-script image: gcr.io/google-containers/startup-script:v1
+        imagePullPolicy: IfNotPresent securityContext: privileged: true env:
+        - name: STARTUP_SCRIPT value: | #! /bin/bash echo 1 >
+          /proc/sys/net/ipv4/netfilter/ip_conntrack_tcp_be_liberal echo done ```
+
```
Comment (DanyC97) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: could you please format the manifest file

**Comment by DanyC97** [comment]
@kbarnard10 maybe you can suggest better rewording here and there

**Comment by anfernee** [comment]
Thanks guys for commenting on the change. Will update. 

One more thing: I don't know what the test fail means. I copied exactly from previous post.
```
9:58:19 AM: Error: Error building site: "/opt/build/repo/content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md:2:1": failed to unmarshal YAML: yaml: mapping values are not allowed in this context
```

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md (hunk 4 above)
Comment (anfernee) [reply content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: It's not exactly what I want to say, but I get your suggestion.

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md (hunk 5 above)
Comment (anfernee) [reply content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: Sorry. My vim `gqG` totally screwed this up.

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+## Summary
```
[the remark this answers is not in the document]
Comment (anfernee) [reply content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: I kept using `gqG` to wrap lines towards the end.. but i didn't notice it wraps quotes too.

File: content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
+---
+title: kube-proxy subtleties: debugging an intermittent connection resets
+date: 2019-03-27
+---
+
+**Author:** Yongkun Gui (Google)
```
Comment (bowei) [inline content/en/blog/_posts/2019-03-27-kube-proxy-subtleties-debugging-an-intermittent-connection-resets.md]: e-mail?

**Comment by anfernee** [comment]
Great. Thanks all!

**Pull request merged**